### PR TITLE
Improve linking to resource class info

### PR DESCRIPTION
**Pull request by rosieyohannan**
# Description
* Add section to executor intro page on resource class with a link to web page
* Convert page to adoc
* Style and formatting improvements

# Reasons
Reports internally that this info is hard to find

# Content Checklist
Please follow our style when contributing to CircleCI docs. Our style guide is here: [https://circleci.com/docs/style/style-guide-overview](https://circleci.com/docs/style/style-guide-overview).

Please take a moment to check through the following items when submitting your PR (this is just a guide so will not be relevant for all PRs) 😸:

- [ ] Break up walls of text by adding paragraph breaks.
- [ ] Consider if the content could benefit from more structure, such as lists or tables, to make it easier to consume.
- [ ] Keep the title between 20 and 70 characters.
- [ ] Consider whether the content would benefit from more subsections (h2-h6 headings) to make it easier to consume.
- [ ] Check all headings h1-h6 are in sentence case (only first letter is capitalized).
- [ ] Is there a "Next steps" section at the end of the page giving the reader a clear path to what to read next?
- [ ] Include relevant backlinks to other CircleCI docs/pages.

### CODE REVIEW

File: jekyll/_cci2/executor-intro.adoc [hunk 1]
```diff
@@ -0,0 +1,343 @@
...
+
+For example:
+
+* Jobs that require Docker images (`docker`) may use an image for Node.js, Python etc. The xref:circleci-images#[pre-built CircleCI Docker image] (convenience images) will help you get started. CircleCI convenience images are not a full operating system, so they generally make building your software more efficient.
+
+* Jobs that require a complete Linux virtual machine (VM) image (`machine`) may use an Ubuntu version supported by the xref:configuration-reference#available-linux-machine-images[list of available machine images].
+
+* Jobs that require a macOS VM image (`macos`) may use an Xcode version. A full list is available on the xref:using-macos#supported-xcode-versions[Using macOS] page.
+
+=== Resource class
```
Comment (rosieyohannan) [inline jekyll/_cci2/executor-intro.adoc:45]: This is the new section

File: jekyll/_cci2/executor-intro.adoc [hunk 2]
```diff
@@ -0,0 +1,346 @@
...
+To access the Linux VM execution environment, use the `machine` executor and specify a Linux image. For a full list of `machine` images, see the link:https://circleci.com/developer/images?imageType=machine[CircleCI Developer Hub]
+
+[.tab.machine.Cloud]
+--
+[,yml]
+----
+jobs:
+  build: # name of your job
+    machine: # executor type
+      image: ubuntu-2004:202010-01 # # recommended linux image - includes Ubuntu 20.04, docker 19.03.13, docker-compose 1.27.4
```
Comment (imlogang) [inline jekyll/_cci2/executor-intro.adoc:86]: Would it be better here to use `ubuntu-2004:current` or `ubuntu-2004:default`? 
We could also do `ubuntu-2204:current` or `ubuntu-2204:default`

File: jekyll/_cci2/executor-intro.adoc [hunk 3]
```diff
@@ -0,0 +1,346 @@
...
+
+workflows:
+  build:
+    jobs:
+      - build-medium
+      - build-large
+----
+--
+
+[.tab.armblock.Server_3]
```
Comment (imlogang) [inline jekyll/_cci2/executor-intro.adoc:296]: Is there a way to specify that, by default, this is AWS only? Our server customers in GCP can't use ARM images.

File: jekyll/_cci2/executor-intro.adoc [hunk 4]
```diff
@@ -0,0 +1,346 @@
...
+--
+[,yml]
+----
+# .circleci/config.yml
+version: 2.1
+
+jobs:
+  build-medium:
+    machine:
+      image: ubuntu-2004:202101-01
```
Comment (imlogang) [inline jekyll/_cci2/executor-intro.adoc:274]: Would it be worth changing these to `ubuntu-2204:current` as well?

**Review by imlogang — APPROVED** [review]

**Pull request merged**